<launch>
  <arg name="to_record" default="false"/>
  <arg name="root_to_record" default="/home/max/sem_keypts_root/"/>
  <arg name="image_topic" default="/gaze/image_resized"/>
  <arg name="odometry_objects" default="[TobiiGlasses, DragonFly1, DragonFly2, DragonFly5]"/>

  <node pkg="sem_keypts_dataset_generator" type="sem_keypts_dataset_generator_node" name="image_processor" output="screen">
      <remap from="~image" to="$(arg image_topic)"/>
      <rosparam param="odometry_objects" subst_value="True">$(arg odometry_objects)</rosparam> 

      <param name="image_width" value="960"/>
      <param name="image_height" value="540"/>

      <param name="~time_offset" value="10"/>

      <param name="~offset_x" value="0"/>
      <param name="~offset_y" value="0"/>
      <param name="~offset_z" value="0.05"/>

      <param name="~offset_roll" value="0.02"/>
      <param name="~offset_pitch" value="-0.296"/>
      <param name="~offset_yaw" value="-0.015"/>

      <param name="~quad_height" value="0.08"/>
      <param name="~quad_width" value="0.2"/>

      <param name="~to_record" value="$(arg to_record)"/>
      <param name="~root" value="$(arg root_to_record)"/>
  </node>
  
</launch>
